File: src/ch/unibas/dmi/dbis/fds/Configuration.java
package ch.unibas.dmi.dbis.fds;

public class Configuration {
	/**
	 * true:  "DEBUG" mode
	 * false: "PRODUCTION" mode
	 */
	private static final boolean DEBUG_MODE = true;


	public static final int INITIAL_NODES;
	public static final int NETWORK_BITS;
	public static final int DEFAULT_FINGER_UPDATE_INTERVAL = 1000;

	static {
		if (DEBUG_MODE) {
			INITIAL_NODES = 5;
			NETWORK_BITS = 4;
		} else {
			INITIAL_NODES = 20;
			NETWORK_BITS = 24;
		}
	}

	public static Network createNetwork() {
		//return new FullyConnectedNetwork(NETWORK_BITS);
		return new ChordNetwork(NETWORK_BITS);
	}
}

File: src/ch/unibas/dmi/dbis/fds/kotlin/Configuration.kt
package ch.unibas.dmi.dbis.fds.kotlin

object Configuration {
    private val DEBUG_MODE = true

    val INITIAL_NODES: Int
    val NETWORK_BITS: Int
    val DEFAULT_FINGER_UPDATE_INTERVAL = 2000

    init {
        if (DEBUG_MODE) {
            INITIAL_NODES = 5
            NETWORK_BITS = 3
        } else {
            INITIAL_NODES = 20
            NETWORK_BITS = 24
        }
    }

    fun createNetwork(): Network {
        //return new FullyConnectedNetwork(NETWORK_BITS);
        return ChordNetwork(NETWORK_BITS)
    }
}
